pkg/controller/knativeserving/openshift/openshift.go

```diff
@@ -167,31 +174,181 @@ func checkVersion(instance *servingv1alpha1.KnativeServing) error {
+			ownerName:      instance.Name,
+			ownerNamespace: instance.Namespace,
+		}),
+	}
+	if err := manifest.Transform(transforms...); err != nil {
+		log.Error(err, "Unable to transform serviceMeshControlPlane manifest")
 		return err
 	}
-	if err := api.Get(context.TODO(), client.ObjectKey{Namespace: "istio-system", Name: "default"}, smmr); err != nil {
-		return checkError(err, instance)
+	if err := manifest.ApplyAll(); err != nil {
+		log.Error(err, "Unable to install serviceMeshControlPlane")
+		return err
 	}
-	found := func(smmr *maistrav1.ServiceMeshMemberRoll) bool {
-		for _, member := range smmr.Status.ConfiguredMembers {
-			if member == "knative-serving" {
-				return true
+	return nil
+}
+
+// appendIfAbsent append namespace to member if its not exist
+func appendIfAbsent(members []string, routeNamespace string) []string {
+	var exist = false
+	for _, val := range members {
+		if val == routeNamespace {
+			exist = true
+			break
+		}
+	}
+	if !exist {
+		members = append(members, routeNamespace)
+	}
+	return members
+}
+
+// installServiceMeshMemberRoll installs ServiceMeshMemberRoll for knative-serving namespace
+func installServiceMeshMemberRoll(instance *servingv1alpha1.KnativeServing) error {
+	smmr := &maistrav1.ServiceMeshMemberRoll{}
+	if err := api.Get(context.TODO(), client.ObjectKey{Namespace: ingressNamespace(instance.Namespace), Name: smmrName}, smmr); err != nil {
+		if apierrors.IsNotFound(err) {
+			smmr.Name = smmrName
+			smmr.Namespace = ingressNamespace(instance.Namespace)
+			smmr.Spec.Members = []string{instance.Namespace}
+			smmr.Labels = map[string]string{
+				ownerName:      instance.Name,
+				ownerNamespace: instance.Namespace,
 			}
+			return api.Create(context.TODO(), smmr)
 		}
-		return false
+		return err
 	}
-	if !found(smmr) {
-		return checkError(errors.New("knative-serving namespace is not a configured member in serviceMeshMemberRoll"), instance)
+	var exist = false
+	// If ServiceMeshMemberRoll already exist than check for knative-serving ns is configured member or not
+	for _, member := range smmr.Status.ConfiguredMembers {
+		if member == instance.Namespace {
+			exist = true
+			break
+		}
+	}
+
+	// if knative-serving ns is not a configured by any chance than update existing ServiceMeshMemberRoll
+	if !exist {
+		smmr.Spec.Members = appendIfAbsent(smmr.Spec.Members, instance.Namespace)
+		return api.Update(context.TODO(), smmr)
+	}
+	return nil
+}
+
+// isServiceMeshMemberRoleReady Checks knative-serving namespace is a configured member or not
+func isServiceMeshMemberRollReady(servingNamespace string) error {
+	smmr := &maistrav1.ServiceMeshMemberRoll{}
+	if err := api.Get(context.TODO(), client.ObjectKey{Namespace: ingressNamespace(servingNamespace), Name: smmrName}, smmr); err != nil {
+		return err
+	}
+	var ready = false
+	for _, member := range smmr.Status.ConfiguredMembers {
+		if member == servingNamespace {
+			ready = true
+			break
+		}
+	}
+	if !ready {
```

Comment:
If you `return nil` in the loop you can always return the error here without additional checks.